.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,97 @@
+name: Deploy to shinyapps.io
+on:
+
+  # run on any tagged push 
+  release:
+    types: [published]
+
+  # run on request (via button in actions menu)
+  workflow_dispatch:
+    inputs:
+      deploy_type:
+        description: Deploy to release or test
+        required: true
+        options:
+          - 'test'
+          - 'release'
+        default: 'test'
+      manual_branch:
```

Comment:
This should not be necessary, by default once the workflow has been pushed to main (or run once), workflow dispatch will be enabled and ask you which branch you want to use, see MetaBayesDTA:
https://github.com/CRSU-Apps/MetaBayesDTA/blob/main/.github/workflows/deploy.yaml#L9
https://github.com/CRSU-Apps/MetaBayesDTA/actions/workflows/deploy.yaml